cmd/kubelet/app/server_others.go

```diff
@@ -28,7 +28,25 @@ func checkPermissions() error {
 	if uid := os.Getuid(); uid != 0 {
 		return fmt.Errorf("kubelet needs to run as uid `0`. It is being run as %d", uid)
 	}
-	// TODO: Check if kubelet is running in the `initial` user namespace.
-	// http://man7.org/linux/man-pages/man7/user_namespaces.7.html
+
+	if err := checkInitialUserNamespace(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// checkInitialUserNamespace checks if kubelet is running in the initial user namespace.
+// http://man7.org/linux/man-pages/man7/user_namespaces.7.html
+func checkInitialUserNamespace() error {
+	uidMap, err := ioutil.ReadFile("/proc/self/uid_map")
+	if err != nil {
+		return err
+	}
+
+	if strings.TrimSpace(string(uidMap)) != "0\t0\t4294967295" {
```

Comment:
How about adding some comments about the expected content of `/proc/self/uid_map`?
For example:
```
// When running in the initial host's user namespace, the `/proc/self/uid_map` file content looks like:
// 0          0 4294967295
```

Reply:
@SataQiu Thank you so much for your review, I will add some comment.